fix: add script to generate core types (from shipped apps)

Antreesy opened the pull request:
### ☑️ Resolves

* Replace manually hardcoded types with ones copied


## 🖌️ UI Checklist

### 🚧 Tasks

- [ ] Script review (maybe there's a better way to do this, with update nextcloud-deps/ocp)
- [ ] Possible conflicts between major versions (might affect Desktop?)

### 🏁 Checklist

- [ ] 🌏 Tested with different browsers / clients:
  - [ ] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [ ] Talk Desktop
  - [ ] Not risky to browser differences / client
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

nickvergessen commented:
Should we run it as an CI step similarly to the OCP update?
https://github.com/nextcloud/spreed/blob/main/.github/workflows/update-nextcloud-ocp.yml

Antreesy commented:
- Added workflow to autogenerate types (based on update-nextcloud/ocp + node)
- Dropped openapi_files_sharing.ts (not used atm and we can test workflow with it)

ShGKme commented on src/types/generate-core-types.sh:
```diff
@@ -0,0 +1,31 @@
+mkdir -p "$TEMP_DIR"
+
+# Find and copy openapi.json files, then translate to ts types
+generate_ts_files() {
+    local full_path=$1           # Full path passed as an argument
+    local new_name="openapi_${full_path#apps/}"
+
+    find "$SERVER_DIR/$full_path" -maxdepth 1 -name "openapi.json" -exec sh -c 'cp "$1" "$2/$3"' _ {} "$TEMP_DIR" "$new_name.json" \;
+	npx openapi-typescript --redocly $TYPES_DIR "$TEMP_DIR/$new_name.json" -t -o "$CORE_TYPES_OUTPUT_DIR/$new_name.ts"
+}
```
And quite hard to read... Does it use `find` to get a file on a specific location? Do we need find in this case (the path is known anyway).

Maybe do it in a bit more lines, but less complex? Like

```sh
    # Find and copy the OpenAPI file
    local openapi_file="$SERVER_DIR/$full_path/openapi.json"
    local temp_file="$TEMP_DIR/$new_name.json"
    
    if [ -f "$openapi_file" ]; then
        cp "$openapi_file" "$temp_file"
    else
        echo "Error: openapi.json not found in $openapi_file"
        return 1
    fi

    # Generate types
    npx ...
```

Antreesy commented:
/backport to stable31

Antreesy commented:
/backport to stable30
